wa/workloads/antutu/__init__.py

```diff
@@ -109,16 +145,98 @@ def extract_scores(self, context, regex_version):
                             result = float('NaN')
                         entry = regex.pattern.rsplit(None, 1)[0]
                         context.add_metric(entry, result, lower_is_better=False)
+        #Calculate group scores if 'CPU' in entry:
+                        if 'CPU' in entry:
+                            cpu.append(result)
+                            cpu_result = sum(cpu)
+                        if 'GPU' in entry:
+                            gpu.append(result)
+                            gpu_result = sum(gpu)
+                        if any([i in entry for i in ['Data', 'Document', 'Image', 'User', 'Video']]):
+                            ux.append(result)
+                            ux_result = sum(ux)
+                        if any([i in entry for i in ['RAM', 'ROM']]):
+                            mem.append(result)
+                            mem_result = sum(mem)
                         expected_results -= 1
         if expected_results > 0:
             msg = "The Antutu workload has failed. Expected {} scores, Detected {} scores."
             raise WorkloadError(msg.format(len(regex_version), expected_results))
 
+        context.add_metric('CPU Total Score', cpu_result, lower_is_better=False)
+        context.add_metric('GPU Total Score', gpu_result, lower_is_better=False)
+        context.add_metric('UX Total Score', ux_result, lower_is_better=False)
+        context.add_metric('MEM Total Score', mem_result, lower_is_better=False)
+
+        #Calculate overall scores
+        overall_result = float(cpu_result + gpu_result + ux_result + mem_result)
+        context.add_metric('Overall Score', overall_result, lower_is_better=False)
```

Comment:
Do these metrics still have meaning for previous versions of the workload? 
Should we only be adding them if that are found during result parsing?

Reply:
Yeah, back to v8 these will always be present. The subtests that make up the groups will be slightly different but all are captured by the logic just above.

Reply:
Ok cool, in which case same as above do we need to add a >= 8?